tests/swarm-deploy/test_swarm_runs.py

```diff
@@ -118,37 +116,73 @@ def test_core_service_running(
     # ID                  NAME                     IMAGE                     NODE                DESIRED STATE       CURRENT STATE            ERROR                       PORTS
     # puiaevvmtbs1        simcore_storage.1       simcore_storage:latest   crespo-wkstn        Running             Running 18 minutes ago
     # j5xtlrnn684y         \_ simcore_storage.1   simcore_storage:latest   crespo-wkstn        Shutdown            Failed 18 minutes ago    "task: non-zero exit (1)"
-    tasks = service.tasks()
-    num_tasks = get_replicas(service_config)
-
-    assert len(tasks) == num_tasks, (
-        f"Expected a {num_tasks} task(s) for '{service_name}', got instead"
-        f"\n{ get_tasks_summary(tasks) }"
-        f"\n{ get_failed_tasks_logs(service, docker_client) }"
-    )
 
-    for i in range(num_tasks):
-        task: Dict[str, Any] = service.tasks()[i]
+    expected_replicas = get_replicas(service_config)
 
-        for n in range(RETRY_COUNT):
-            task = service.tasks()[i]
+    # assert number of slots equals number of expected services
+    tasks = service.tasks()
+    slots: Set[int] = {t["Slot"] for t in tasks}
+    slot_count = len(slots)
+
+    assert slot_count == expected_replicas, (
+        f"Expected to have {expected_replicas} slot(s), "
+        f"instead {slot_count} slot(s) were found."
+        f"\n{get_tasks_summary(tasks)}"
+        f"\n{get_failed_tasks_logs(service, docker_client)}"
+    )
 
-            if task["Status"]["State"].upper() in SWARM_TASK_PRE_STATES:
-                print(
-                    "Waiting [{}/{}] ...\n{}".format(
-                        n, RETRY_COUNT, get_tasks_summary(tasks)
-                    )
+    def _get_tasks_by_slots() -> Dict[int, Deque[Dict[str, Any]]]:
+        results: Dict[int, Deque[Dict[str, Any]]] = {}
+
+        for task in service.tasks():
+            slot = task["Slot"]
+            if slot not in results:
+                results[slot] = deque()
+            results[slot].append(task)
+
+        return results
+
+    def _are_any_tasks_running(slot_tasks: Deque[Dict[str, Any]]) -> bool:
+        running_tasks = [
+            t for t in slot_tasks if t["Status"]["State"].upper() == "RUNNING"
+        ]
+        return len(running_tasks) > 0
+
+    # check at least one service in per slot is in running mode else raise error

```

Comment:
this comment is very confusing. Please check the terminology.
A service -> creates a bunch of tasks (that are analogous to slots), which in fine are containers. not the contrary.